great_expectations/util.py

```diff
@@ -122,6 +125,11 @@ def singularize(plural_ge_noun):
         )
 
 
+def camel_to_snake(name):
+    name = p1.sub(r"\1_\2", name)
+    return p2.sub(r"\1_\2", name).lower()
```

Comment:
Could we type annotate the args and return here? Also, could we add a few unit tests? I trust your regex skills a lot more than I do mine but I think a few tests to gain coverage would be great!

Reply:
Oops now seeing you just ported this over - will defer to you here.

Reply:
Just moved as-is from great_expectations/expectations/expectation.py